webcomponents/tool/src/main/frontend/js/tasks/sakai-tasks-create-task.js

```diff
@@ -132,16 +132,26 @@ export class SakaiTasksCreateTask extends SakaiDialogContent {
 
       const datePicker = this.shadowRoot.getElementById("due");
```

Comment:
Hi Earle, I think the selector used is correct. However, I suspect the issue is that the first time a user logs in, the element with id="due" has not been rendered yet, causing the error to occur. It's possible that this issue is related to the browser cache, as the error doesn't seem to happen if the user logs out and logs back in.

<img width="1244" alt="Screenshot 2023-05-04 at 8 46 20 PM" src="https://user-images.githubusercontent.com/102482288/236362906-52cf45f7-1fb5-4109-b151-1f034e4d8a2c.png">

Reply:
The issue occurs because in the constructor function of ```<sakai-tasks-create-task>```, the ```this.task``` assignment triggers the ```set task()``` function. Inside the ```set task()``` function, there is a selector that attempts to find the element with id="due" within the ```<sakai-tasks-create-task>``` element. However, during the first login, the ```<sakai-tasks-create-task>``` element has not been rendered yet, resulting in errors.